martin-mbtiles/src/mbtiles.rs

```diff
@@ -268,6 +274,31 @@ impl Mbtiles {
         }
         Ok(None)
     }
+
+    pub async fn detect_type<T>(&self, conn: &mut T) -> MbtResult<Type>
+    where
+        for<'e> &'e mut T: SqliteExecutor<'e>,
+    {
+        let is_tiles_tables_type: i32 =
+            query(include_str!("queries/is_tile_tables_mbtiles_type.sql"))
+                .fetch_one(&mut *conn)
+                .await?
+                .get("is_valid");
+        if 1 == is_tiles_tables_type {
```

Comment:
```suggestion
        if is_tiles_tables_type == 1 {
```

constants are always after the variable for readibility. The `const == var` is needed for c/c++ because it has no proper "boolean" type, and it allows `if (var = 1) { ... }` syntax (although it does give a warning)